File: python/euler1_4.py
#!/usr/bin/python3
# Euler1 in Python

from functools import reduce

myMap = lambda size: map(lambda x: x, range(size))
myFilter = lambda ints: filter(lambda x: x%3==0 or x%5==0, ints)
myReduce = lambda filtered: reduce(lambda x,y: x+y, filtered)

print ("Euler1 =", myReduce(myFilter(myMap(1000))))

File: lua/euler1_4.lua
#!/usr/bin/lua5.4
-- Euler1 in Lua

function euler1(n)
    result = 0

    for i = 0, math.floor(n/3) do
        result = result + (i * 3)
    end
    for i = 0, math.floor(n/5) do
        result = result + (i * 5)
    end
    for i = 0, math.floor(n/15) do
        result = result - (i * 15)
    end

    return result
end

print (string.format("Euler1 = %d", euler1(999)))
